### DISCUSSION TITLE
Why does Prism for WPF does not have an IInitialize interface?

### DISCUSSION THREAD

**Discussion by MariusHBG** [PrismLibrary/Prism#2485] [WPF]
Hi,

Prism for Xamarin has the option to handle view model initialization by using the IInitialize or IInitializeAsync interfaces. This gives access to the navigation parameters upon first navigation to a view model. 
I would like to use this behavior in a WPF app, but Prism for WPF does not include the IInitialize or IInitializeAsync interfaces, and the INavigationAware interface does not provide similar functionality. 

Why is this the case? Is this a technical limitation? What is the recommended approach for a WPF app to achieve this behavior?

Greetings

**Comment by brianlagunas** [comment]
WPF doesn't need this interface. The IInitialize was added to XF because of how XF works with deep linking. In WPF, you just use OnNavigatedTo.
